src/main/webapp/static/js/init.js

```diff
@@ -1,10 +1,31 @@
 (function($){
   $(function(){
-
+    var web3Ready = false;
     $('.button-collapse').sideNav();
     $('#navButton').sideNav();
     
     $('select').material_select();
+    $('#web3').click(function() {
+      if (!web3Ready){
+        if (window.ethereum) {  
+          window.web3 = new Web3(ethereum);  
+          window.ethereum.enable()
+        } else if (window.web3) {
+          window.web3 = new Web3(web3.currentProvider)
+        } else {
+          console.log('No Metamask (or other Web3 Provider) installed')
+          return
+        }
+        web3Ready = true;
+        return
+      }
+      ethAddress = web3.currentProvider.selectedAddress;
```

Comment:
Shouldn't this be `window.web3` instead of `web3`?

Reply:
Looks like it could be either `web3` or `window.web3` :D see: https://stackoverflow.com/questions/57529226/window-properties-without-window

Reply:
@hhio618 Ok. I was asking in case using `web3` instead of `window.web3` would cause the code to crash when the user does not have support for Web3 in their browser.